<div align="center"> <b> <span style='color:blue'> <h1> IS 362 - 311 SERVICE REQUESTS 2015 </h1> </span> </b> </div>
<div align="center"> <b> <span style='color:blue'> <h2> FINAL PROJECT </h2> </span> </b> </div>

For my final project I plan to demonstrate all that I have learned throughout the course. I will be using data from [NYC OpenData](https://opendata.cityofnewyork.us/). Utilizing the available data, I have decided to use [311 Service Requests](https://data.cityofnewyork.us/dataset/311-Service-Requests-From-2015/57g5-etyj) for 2015. I have filtered the data to shorten the amount of wait time for results as the dat is very large. <br /><br /> With the available data I plan to show:
<ul>
<li> What was the number #1 service request handled by 311 in 2015? </li>
<li> What season received the most service requests? Name a few of the SR’s </li>
<li> What borough made the most service request in 2015? </li>
<li> What was the total of service requests in 2015? </li>
<li> How many service requests remain open? How many have been closed? </li>
</ul>

I plan to have a pie graph to display the seasons and possibly the borough with most service requests. I also plan to include some images <i>(if I can figure it out)</i> and some other types of graphs to display more info on service requests for 2015.

<b> We can begin by importing an image, the necessary libraries, and reading the data. </b>

In [55]:
import pandas as pd
import numpy as np
from IPython.display import Image

Image(url='http://www1.nyc.gov/assets/home/images/social/applications/NYC311Logo-fff200-480px-programs.png')

In [57]:
#reading the data
url = 'https://raw.githubusercontent.com/eliz79/IS362_Final_Project.-/master/311_SR_2015_Filtered.csv'
ser_req = pd.read_csv(url)
ser_req.head(10)

Unique Key     Created Date     Closed Date Agency  \
0    31015465   7/6/2015 10:58  7/22/2015 1:07    DCA   
1    30997660   7/3/2015 13:26  7/3/2015 14:08   NYPD   
2    31950223   11/9/2015 3:55  11/9/2015 8:08   NYPD   
3    31000038    7/3/2015 2:18   7/3/2015 7:54   NYPD   
4    30995614    7/4/2015 0:03   7/4/2015 3:33   NYPD   
5    31042454    7/9/2015 0:00  7/20/2015 0:00  DOHMH   
6    31043076   7/9/2015 12:04             NaN    DPR   
7    31037751    7/9/2015 0:00             NaN  DOHMH   
8    31298553  8/12/2015 11:09  8/28/2015 1:06    DCA   
9    31492526   9/9/2015 21:59  9/9/2015 23:17   NYPD   

                               Agency Name                 Complaint Type  \
0           Department of Consumer Affairs             Consumer Complaint   
1          New York City Police Department                        Vending   
2          New York City Police Department               Blocked Driveway   
3          New York City Police Department             Noise - Commercial   
4          New York City Police Department        Noise - Street/Sidewalk   
5  Department of Health and Mental Hygiene                 Standing Water   
6       Department of Parks and Recreation  Root/Sewer/Sidewalk Condition   
7  Department of Health and Mental Hygiene                 Standing Water   
8           Department of Consumer Affairs             Consumer Complaint   
9          New York City Police Department        Noise - Street/Sidewalk   

                    Descriptor                 Location Type  Incident Zip  \
0              Demand for Cash                           NaN       11360.0   
1           In Prohibited Area    Residential Building/House       10019.0   
2                    No Access               Street/Sidewalk       10453.0   
3             Loud Music/Party           Club/Bar/Restaurant       11372.0   
4                 Loud Talking               Street/Sidewalk       11216.0   
5        Other - Explain Below                         Other           NaN   
6  Trees and Sidewalks Program                        Street       10469.0   
7             Puddle in Ground  3+ Family Apartment Building       10016.0   
8      Damaged/Defective Goods                           NaN       11420.0   
9                 Loud Talking               Street/Sidewalk       11238.0   

           Incident Address              City  Status  \
0          27-16 203 STREET           BAYSIDE  Closed   
1    200 CENTRAL PARK SOUTH          NEW YORK  Closed   
2         1993 GRAND AVENUE             BRONX  Closed   
3  84-16 NORTHERN BOULEVARD   JACKSON HEIGHTS  Closed   
4        1057 BERGEN STREET          BROOKLYN  Closed   
5                       NaN               NaN  Closed   
6      3344 PEARSALL AVENUE             BRONX    Open   
7          379 THIRD AVENUE          NEW YORK    Open   
8         127-19 111 AVENUE  SOUTH OZONE PARK  Closed   
9     238 SAINT JAMES PLACE          BROOKLYN  Closed   

                              Resolution Description    Borough  
0  The Department of Consumer Affairs mailed you ...     QUEENS  
1  The Police Department responded to the complai...  MANHATTAN  
2  The Police Department responded to the complai...      BRONX  
3  The Police Department responded to the complai...     QUEENS  
4  The Police Department responded and upon arriv...   BROOKLYN  
5  The Department of Health and Mental Hygiene wi...     QUEENS  
6  The Department of Parks and Recreation usually...      BRONX  
7  The Department of Health and Mental Hygiene wi...  MANHATTAN  
8  The Department of Consumer Affairs mailed you ...     QUEENS  
9  The Police Department responded to the complai...   BROOKLYN

<b> I decided to replace NaN values with '0'. </b>

In [58]:
ser_req.replace(np.NaN, 0)

Unique Key     Created Date       Closed Date Agency  \
0      31015465   7/6/2015 10:58    7/22/2015 1:07    DCA   
1      30997660   7/3/2015 13:26    7/3/2015 14:08   NYPD   
2      31950223   11/9/2015 3:55    11/9/2015 8:08   NYPD   
3      31000038    7/3/2015 2:18     7/3/2015 7:54   NYPD   
4      30995614    7/4/2015 0:03     7/4/2015 3:33   NYPD   
5      31042454    7/9/2015 0:00    7/20/2015 0:00  DOHMH   
6      31043076   7/9/2015 12:04                 0    DPR   
7      31037751    7/9/2015 0:00                 0  DOHMH   
8      31298553  8/12/2015 11:09    8/28/2015 1:06    DCA   
9      31492526   9/9/2015 21:59    9/9/2015 23:17   NYPD   
10     31495596   9/9/2015 12:12  12/15/2015 14:07    DPR   
11     31593923  9/22/2015 13:50                 0    DPR   
12     31593417  9/22/2015 13:12   9/25/2015 11:20    DOB   
13     31593599  9/22/2015 15:07                 0    DPR   
14     30502370  4/28/2015 18:26   4/28/2015 19:29   NYPD   
15     30498881  4/28/2015 17:54   10/8/2015 16:30    DOT   
16     31524474  9/13/2015 13:35   9/13/2015 22:02   NYPD   
17     31527345  9/13/2015 21:04    9/14/2015 1:51   NYPD   
18     31006258   7/4/2015 16:57     9/3/2015 6:26  DOHMH   
19     30668699  5/21/2015 19:01   5/21/2015 21:56   NYPD   
20     31060994   7/13/2015 1:14    7/13/2015 7:20   NYPD   
21     31181885  7/28/2015 10:16   7/28/2015 11:52   NYPD   
22     30673749  5/21/2015 20:40   5/21/2015 23:19   NYPD   
23     30671660   5/21/2015 4:43    5/21/2015 6:18   NYPD   
24     30670824  5/21/2015 18:31   5/21/2015 21:45   NYPD   
25     31435374  8/31/2015 15:35    9/1/2015 12:00   DSNY   
26     31591848  9/22/2015 20:51   2/16/2016 13:44    DPR   
27     31593561  9/22/2015 10:29  10/13/2015 11:01    DPR   
28     31458870   9/3/2015 14:13    9/19/2015 1:06    DCA   
29     31591347  9/22/2015 13:06  11/11/2015 13:34    DPR   
..          ...              ...               ...    ...   
119    31612632  9/25/2015 11:44   9/25/2015 18:15   NYPD   
120    31714249  10/8/2015 13:12   10/8/2015 14:29   NYPD   
121    31710637  10/8/2015 14:59   10/8/2015 20:00   NYPD   
122    31611944   9/25/2015 7:21    9/25/2015 7:28   NYPD   
123    31613802  9/25/2015 11:43   9/29/2015 15:32    DOT   
124    31613948  9/25/2015 11:47   9/25/2015 21:01   NYPD   
125    31527101   9/13/2015 8:58    9/13/2015 9:30   NYPD   
126    31526392  9/13/2015 17:16   9/13/2015 18:40   NYPD   
127    31526296  9/13/2015 22:20   9/13/2015 22:53   NYPD   
128    31480765   9/7/2015 21:44    9/7/2015 22:19   NYPD   
129    31524842  9/13/2015 15:55   9/13/2015 17:09   NYPD   
130    31526951  9/13/2015 12:50   9/13/2015 15:08   NYPD   
131    31527799  9/13/2015 22:08   9/13/2015 23:04   NYPD   
132    30487472  4/26/2015 23:13    4/27/2015 3:47   NYPD   
133    30538581    5/4/2015 1:45     5/4/2015 2:24   NYPD   
134    30485019   4/27/2015 1:00    4/27/2015 2:35   NYPD   
135    30136906   3/9/2015 15:50    3/9/2015 21:05   NYPD   
136    30601523   5/12/2015 9:11   5/15/2015 11:46    DOT   
137    30898443  6/20/2015 12:43   6/20/2015 15:24   NYPD   
138    31708315  10/8/2015 10:09   10/11/2015 8:41    HPD   
139    30458302  4/22/2015 12:53   4/22/2015 13:09   NYPD   
140    30472249  4/24/2015 20:35   4/24/2015 21:17   NYPD   
141    30897402   6/21/2015 0:54    6/21/2015 1:59   NYPD   
142    29989497  2/19/2015 13:11    2/25/2015 4:22    DOT   
143    29988672  2/19/2015 16:37    2/25/2015 4:22    DOT   
144    30895618  6/20/2015 22:35   6/20/2015 23:03   NYPD   
145    30894796  6/20/2015 16:00   6/20/2015 17:24   NYPD   
146    30895197  6/20/2015 12:37    9/8/2015 14:54    TLC   
147    31611640  9/25/2015 19:57   10/11/2015 1:06    DCA   
148    30895161  6/20/2015 22:04   6/20/2015 22:53   NYPD   

                                           Agency Name  \
0                       Department of Consumer Affairs   
1                      New York City Police Department   
2                      New York City Police Depa

<b> I decided to take a look at the data types of each column to understand what types I will be working with and may need to change if necessary. </b>

In [59]:
ser_req.dtypes

Unique Key                  int64
Created Date               object
Closed Date                object
Agency                     object
Agency Name                object
Complaint Type             object
Descriptor                 object
Location Type              object
Incident Zip              float64
Incident Address           object
City                       object
Status                     object
Resolution Description     object
Borough                    object
dtype: object

<b> We now can begin to answer the questions from above. <br /> <br /> 
Q1. What was the #1 service request handled by 311 in 2015? </b>

In [60]:
ser_req['Complaint Type'].value_counts()

Illegal Parking                  29
Blocked Driveway                 24
Noise - Street/Sidewalk          22
Street Condition                 12
Noise - Commercial                9
Consumer Complaint                5
Overgrown Tree/Branches           4
Derelict Vehicle                  4
Root/Sewer/Sidewalk Condition     3
Damaged Tree                      3
Sidewalk Condition                2
Vending                           2
Benefit Card Replacement          2
Standing Water                    2
Broken Muni Meter                 2
Noise - Park                      2
Homeless Encampment               2
Traffic                           2
Noise - Vehicle                   2
HEAT/HOT WATER                    2
Dead Tree                         1
Maintenance or Facility           1
Rodent                            1
Ferry Complaint                   1
Food Establishment                1
Construction                      1
Illegal Tree Damage               1
Taxi Complaint              

<b> We can view from the results above that ILLEGAL PARKING complaints was the complaint type with the highest outcome with a total of 29 requests. </b>
<br /> <br />
<b> Now I will create a chart in Tableau to display for the above results. </b>

In [61]:
chart = 'https://raw.githubusercontent.com/eliz79/IS362_Final_Project.-/master/Complaint%20Types.csv'
chart_ser_req = pd.read_csv(chart)
chart_ser_req.head()

Complaint Types   Totals 
0  Illegal Parking                     29
1     Blocked Driveway                 24
2     Noise - Street/Sidewalk          22
3      Street Condition                12
4         Noise - Commercial            9

<b>I used the results above to create a chart <i>(.csv file)</i> and saved the results. Take a look at the [Complaint Types Chart](https://public.tableau.com/profile/erica3118#!/vizhome/ComplaintTypes/Sheet1?publish=yes) to view the chart in Tableau. From the chart we are able to view that Illegal Parking was the service request taken most in 2015.</b>
<br /> <br />

<b> We now can continue to answer the next question. <br />
Question #2: Based on the data given, what season recieved the most service requests in 2015 ? </b>

In [62]:
date = ser_req.groupby('Created Date').count()
date.head()

Unique Key  Closed Date  Agency  Agency Name  \
Created Date                                                     
10/1/2015 7:02             1            1       1            1   
10/19/2015 13:32           1            1       1            1   
10/19/2015 7:51            1            1       1            1   
10/8/2015 10:09            1            1       1            1   
10/8/2015 13:12            2            2       2            2   

                  Complaint Type  Descriptor  Location Type  Incident Zip  \
Created Date                                                                
10/1/2015 7:02                 1           1              1             1   
10/19/2015 13:32               1           1              0             1   
10/19/2015 7:51                1           1              1             1   
10/8/2015 10:09                1           1              1             1   
10/8/2015 13:12                2           2              2             2   

                  Incident Address  City  Status  Resolution Description  \
Created Date                                                               
10/1/2015 7:02                   1     1       1                       1   
10/19/2015 13:32                 1     1       1                       1   
10/19/2015 7:51                  1     1       1                       1   
10/8/2015 10:09                  1     1       1                       1   
10/8/2015 13:12                  2     2       2                       2   

                  Borough  
Created Date               
10/1/2015 7:02          1  
10/19/2015 13:32        1  
10/19/2015 7:51         1  
10/8/2015 10:09         1  
10/8/2015 13:12         2

<b> Based on the results, the date of October 8th recieved the most service requests. It got a total of 2 complaints for that day. With such conclusions, we can say that the FALL season recieved the most service requests in 2015 as October is in the Fall season. Take a look at the [SEASON'S CHART](https://public.tableau.com/profile/erica3118#!/vizhome/DateChart/Sheet1?publish=yes). <br /> <br />
Now let's move unto the next question. <br />  
Q3: What borough submitted the most service requests in 2015? </b>

In [63]:
borough = ser_req.groupby('Borough').count()
borough.head()

Unique Key  Created Date  Closed Date  Agency  Agency Name  \
Borough                                                                     
BRONX                  29            29           27      29           29   
BROOKLYN               40            40           39      40           40   
MANHATTAN              30            30           29      30           30   
QUEENS                 40            40           38      40           40   
STATEN ISLAND           6             6            6       6            6   

               Complaint Type  Descriptor  Location Type  Incident Zip  \
Borough                                                                  
BRONX                      29          29             29            29   
BROOKLYN                   40          40             40            40   
MANHATTAN                  30          28             28            30   
QUEENS                     40          40             35            39   
STATEN ISLAND               6           6              6             6   

               Incident Address  City  Status  Resolution Description  
Borough                                                                
BRONX                        22    29      29                      28  
BROOKLYN                     31    40      40                      40  
MANHATTAN                    20    30      30                      30  
QUEENS                       33    39      40                      40  
STATEN ISLAND                 4     6       6                       6

<b> From the results we can see that BROOKLYN and QUEENS were the two boroughs with most service requests in 2015 with a total of 40 requests each. <br /><br /> Again, we can view the results in the [Borough Service Request Chart](https://public.tableau.com/profile/erica3118#!/vizhome/BoroughServiceRequests-2015/Sheet1?publish=yes) </b>
<br /><br />
<b> We can proceed unto the next question. </b> <br /><br />
<b> Question #3: What what the total of service requests in 2015? </b>

In [64]:
ser_req['Complaint Type'].count()

149

<b> There were a total of 149 service requests in 2015. But how many of those are closed & how many remain open? </b>

In [65]:
ser_req['Status'].value_counts()

Closed     143
Open         5
Pending      1
Name: Status, dtype: int64

<b> There are 143 CLOSED and 5 service request which remained OPEN. ONE service request remained pending. Let's take a look at the [2015 - Total Service Requests Chart](https://public.tableau.com/profile/erica3118#!/vizhome/TotalServiceRequests-2015/Sheet1?publish=yes)</b>

<div align="center"> <b> <span style='color:blue'> <h1> THANK YOU </h1> </span> </b> </div>